org.opentosca.bus/org.opentosca.bus.application.api.soaphttp/src/main/java/org/opentosca/bus/application/api/soaphttp/route/Route.java

```diff
@@ -41,26 +37,19 @@ public void configure() throws Exception {
             + "&serviceName={http://opentosca.org/appinvoker/}AppInvokerSoapWebServiceService&portName="
             + PORT.toString() + "&dataFormat=PAYLOAD&loggingFeatureEnabled=true";
 
-        final ValueBuilder APP_BUS_ENDPOINT =
-            new ValueBuilder(method(ApplicationBusServiceHandler.class, "getApplicationBusRoutingEndpoint"));
-        final Predicate APP_BUS_ENDPOINT_EXISTS = PredicateBuilder.isNotNull(APP_BUS_ENDPOINT);
-
         final ClassLoader cl = org.opentosca.bus.application.api.soaphttp.model.ObjectFactory.class.getClassLoader();
         final JAXBContext jc = JAXBContext.newInstance("org.opentosca.bus.application.api.soaphttp.model", cl);
         final JaxbDataFormat jaxb = new JaxbDataFormat(jc);
 
         final Processor requestProcessor = new RequestProcessor();
         final Processor responseProcessor = new ResponseProcessor();
 
-        from(SOAP_ENDPOINT).unmarshal(jaxb).process(requestProcessor).choice().when(APP_BUS_ENDPOINT_EXISTS)
-            //FIXME this recipientList should be replaced with a directly injected service reference
-            .recipientList(APP_BUS_ENDPOINT).to("direct:handleResponse").endChoice().otherwise()
-            .to("direct:handleException");
+        from(SOAP_ENDPOINT).unmarshal(jaxb).process(requestProcessor)
+                           .to("direct-vm:org.opentosca.bus.application.service").to("direct:handleResponse");

```

Comment:
the recipient list is gone? wasn't there a problem in the past, which was elegantly solved with the list ?

Reply:
It should be working like this now.